pkg/network/network.go

```diff
@@ -167,7 +169,11 @@ func ValidateUserNetworkIsAvailable(config *config.Config, userNet *net.IPNet) e
 
 // RemoveNetwork removes a given network by name.  If the network has container associated with it, that
 // must be handled outside the context of this.
-func RemoveNetwork(config *config.Config, name string) error {
+func RemoveNetwork(config *config.Config, name string, lockPath string) error {
+	// Acquire a lock for network removal
+	if _, err := storage.GetLockfile(filepath.Join(lockPath, LockFileName)); err != nil {
+		return err
+	}
```

Comment:
Same issue as above.

I think it's _very_ risky to use the lock here _and_ outside of the package. It suggests that code outside of the package (i.e., `createMacVLAN` and `createBridge`) that needs the lock should rather live inside the package. Callers shouldn't need to worry about concurrency issues; the risk of missing to lock or running into deadlocks is high.

Reply:
yeah, i waffled on this as well.  I'm not sure I would categorize the risk similar but it is a fair comment.  I'll consider a move.